### DISCUSSION TITLE
I am trying to play an adaptive bitrate streaming video (HLS) in a cargo-leptos SSR'd site. What would be the easiest way to do this?

### DISCUSSION THREAD

**Discussion by saikatdas0790** [leptos-rs/leptos#2088] [Q&A]
I have a URL to a manifest file that looks like this:
https://customer-2p3jflss4r4hmpnz.cloudflarestream.com/4c5064176ce24f29b46f8c614acbe6e1/manifest/video.m3u8

What would the easiest path be to stream video to the browser using Leptos for an HLS stream?

**Comment by gbj** [comment]
I genuinely have no idea, but I suspect the answer would be the same as the JavaScript answer in this case — as I understand it HLS is not supported across browsers so requires some third-party library. But that's about the extent of my knowledge.

**Reply by saikatdas0790** [reply]
Yes, that is my conclusion as well. I really don't want to write my own HLS player [as suggested by many](https://web.dev/articles/media-frameworks#benefits_of_using_a_framework)

And the only WASM specific library for HLS playback [doesn't seem to support](https://github.com/peaBerberian/wasp-hls/issues/7) usage from a Rust codebase directly

So, I'm kinda lost on the right way forward. Might have to fall back to building a JS webapp which makes me sad. 🥹
Let's see..
